### DISCUSSION TITLE
part-ii/2014-102/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#207] [Question comments]
Part II 2014 Number Theory: Paper 4, Section II, F 

https://questions.tripos.org/part-ii/2014-102/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

The first few parts are bookwork.

If $p \equiv 1 \pmod{4}$, then $(\frac{-1}{p}) = +1$ (this follows from Euler's criterion), so the quadratic residues in $\{1, 2, \dots, p - 1\}$ are of the form $\{a_1, p - a_1, a_2, p - a_2, \dots\}$ and similarly for the quadratic non-residues. There are $\frac{p-1}{2}$ quadratic residues (and the same number of quadratic non-residues), and we can pair up terms that sum to $p$ for both of these sets (so we have $\frac{p-1}{4}$ pairs of quadratic residues; this is an integer because $p \equiv 1 \pmod{4}$). Therefore the sum of the quadratic residues and the sum of the quadratic non-residues are both equal to $p(\frac{p-1}{2})$, so they are equal.

Note that $7$ is not a quadratic residue $\pmod{7}$ by definition, so we assume that $p \neq 7$. We can easily find by trying all of the possibilities that the quadratic residues $\pmod{7}$ are $1$, $2$ and $4$. We use the law of quadratic reciprocity, and split into two cases based on whether $p \equiv 1 \pmod{4}$ or $p \equiv 3 \pmod{4}$.

If $p \equiv 1 \pmod{4}$, then $(\frac{7}{p}) = (\frac{p}{7})$ by the law of quadratic reciprocity, and this is equal to $1$ if (and only if) $p \equiv 1, 2 \text{ or } 4 \pmod{7}$. Combining this with the condition $\pmod{4}$ (and the Chinese Remainder Theorem), we get the possibilities $p \equiv 1, 9, 25 \pmod{28}$.

If instead $p \equiv 3 \pmod{4}$, then $(\frac{7}{p}) = -(\frac{p}{7})$, which is equal to $1$ if and only if $p \equiv 3, 5 \text{ or } 6 \pmod{7}$.  Combining this with the condition $\pmod{4}$ and the Chinese Remainder Theorem, we get the possibilities $p \equiv 3, 19, 27 \pmod{28}$.

Therefore the primes $p$ for which $7$ is a quadratic residue are exactly those such that

$$p \equiv 1, 3, 9, 19, 25, 27 \pmod{28}$$

</details>
